### ISSUE TITLE
PHPCS v4 compatibility bug

### ISSUE THREAD

**Issue by dereuromark** [slevomat/coding-standard#1790]
# Bug Report: AnnotationHelper::fixAnnotation Changes Indentation from Spaces to Tabs

## Summary
When using `AnnotationHelper::fixAnnotation()` to modify a PHPDoc annotation in a union type containing an array type hint, the helper changes the indentation of the entire doc comment from spaces to tabs, even when the original file uses spaces.

## Environment
- SlevomatCodingStandard version: dev-phpcs4 (commit fe765b33fb64f7904792aff481c91d8ed13ca571)
- PHP_CodeSniffer version: ^4.0.0
- PHP version: 8.4.12

## Steps to Reproduce

1. Create a PHP file with space-indented PHPDoc comment:
```php
<?php
class Test {
    /**
     * @return \ArrayObject<string>|int[]
     */
    protected function getArrayObjectOfStringsOrArrayOfInts()
    {
        return $this->foo();
    }
}
```

Note: The file uses 4 spaces for indentation (not tabs).

2. Use `AnnotationHelper::fixAnnotation()` to replace `int[]` with `array<int>` in the union type.

3. Apply the fix using `FixerHelper::change()`.

## Expected Behavior
The fixed doc comment should preserve the original indentation style (spaces):
```php
    /**
     * @return \ArrayObject<string>|array<int>
     */
```

## Actual Behavior
The fixed doc comment is changed to use tabs:
```php
    /**
	 * @return \ArrayObject<string>|array<int>
	 */
```

Note the second and third lines now use tabs (shown as `^I` in cat -A output).

## Code Path
The issue occurs when:
1. `AnnotationHelper::fixAnnotation()` is called to fix a union type annotation
2. The fixed content is applied using `FixerHelper::change()`

Example code from our sniff:
```php
$fixedDocComment = AnnotationHelper::fixAnnotation(
    $parsedDocComment,
    $annotation,
    $unionTypeNode,
    $genericTypeNode,
);

// Later...
$phpcsFile->fixer->beginChangeset();
FixerHelper::change(
    $phpcsFile,
    $parsedDocComment->getOpenPointer(),
    $parsedDocComment->getClosePointer(),
    $fixedDocComment,
);
$phpcsFile->fixer->endChangeset();
```

## Impact
This causes issues in codebases that enforce space-only indentation, as the fixer changes the indentation style of doc comments it modifies.

## Possible Solution
`AnnotationHelper::fixAnnotation()` should detect and preserve the original indentation style of the doc comment, or provide an option to specify the desired indentation style.

## Test Case
Input file (with spaces):
```
    /**
     * @return \ArrayObject<string>|int[]
     */
```

After running through AnnotationHelper::fixAnnotation + FixerHelper::change:
```
    /**
	 * @return \ArrayObject<string>|array<int>
	 */
```

The indentation characters have changed from spaces to tabs.

**Comment by dereuromark** [comment]
A possible fix:
```diff
diff --git a/SlevomatCodingStandard/Helpers/AnnotationHelper.php b/SlevomatCodingStandard/Helpers/AnnotationHelper.php
index ebce8e72..14e15ba0 100644
--- a/SlevomatCodingStandard/Helpers/AnnotationHelper.php
+++ b/SlevomatCodingStandard/Helpers/AnnotationHelper.php
@@ -25,7 +25,12 @@ use PHPStan\PhpDocParser\Ast\Type\ObjectShapeItemNode;
 use PHPStan\PhpDocParser\Ast\Type\ObjectShapeNode;
 use PHPStan\PhpDocParser\Ast\Type\UnionTypeNode;
 use function in_array;
+use function preg_match;
+use function preg_replace_callback;
 use function sprintf;
+use function str_repeat;
+use function strlen;
+use function strpos;
 use function strtolower;
 
 /**
@@ -171,11 +176,33 @@ class AnnotationHelper
 			}
 		}
 
-		return PhpDocParserHelper::getPrinter()->printFormatPreserving(
+		$result = PhpDocParserHelper::getPrinter()->printFormatPreserving(
 			$newPhpDocNode,
 			$parsedDocComment->getNode(),
 			$parsedDocComment->getTokens(),
 		);
+
+		// Detect original indentation style from the tokens
+		$originalContent = $parsedDocComment->getTokens()->getContentBetween(0, $parsedDocComment->getTokens()->getTokenCount());
+		$usesSpaces = false;
+		$indentChar = "\t";
+
+		// Check if original uses spaces for indentation by looking at lines after /**
+		if (preg_match('/\n( +)\*/', $originalContent, $matches)) {
+			$usesSpaces = true;
+			$indentChar = ' ';
+		}
+
+		// If original used spaces but result has tabs, convert back to spaces
+		if ($usesSpaces && strpos($result, "\n\t") !== false) {
+			// Replace tabs with spaces, preserving the same visual indentation
+			// Typically 1 tab = 4 spaces in PHPDoc context
+			$result = preg_replace_callback('/\n(\t+)/', function ($matches) {
+				return "\n" . str_repeat(' ', strlen($matches[1]) * 4);
+			}, $result);
+		}
+
+		return $result;
 	}
 
 	/**
```

**Comment by dereuromark** [comment]
I dont quite get how this wasnt an issue with this library and phpcs v3
Since the dependencies otherwise didnt change.

**Comment by jrfnl** [comment]
@dereuromark Just thinking - you might want to add yourself to the [community cc-list](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/4.x/.github/community-cc-list.md?plain=1) to get a heads-up when proposals are posted which affect external standards (Tokenizer) or integrations...?

**Comment by dereuromark** [comment]
Fairly certain this is a bug in the helper and how v4 handles it.
https://github.com/php-collective/code-sniffer/pull/34/commits/fc992f73139d7c9a149b08aeb61e990f4c06d58f fixes it for me though for now, so I am not interested in pursuing this further.
Feel free to close.
